src/lib.rs

```diff
@@ -442,17 +480,30 @@ impl<D: tb::Bool, T: tb::Bool> InvoiceBuilder<D, tb::False, T> {
 
 impl<D: tb::Bool, H: tb::Bool> InvoiceBuilder<D, H, tb::False> {
 	/// Sets the timestamp. `time` is a UNIX timestamp.
-	pub fn timestamp(mut self, time: u64) -> InvoiceBuilder<D, H, tb::True> {
-		self.timestamp = Some(time);
+	pub fn timestamp_raw(mut self, time: u64) -> InvoiceBuilder<D, H, tb::True> {
```

Comment:
I think the safety argument I made would require me to remove this function since it encourages non-semantic type usage. Some tests will become a bit more painful, but that should be ok.